packages/core/src/styles/layout/_type-horizontal.scss

```diff
@@ -2,135 +2,134 @@
-      justify-content: stretch;
+@mixin generateHorizontalStretch($breakpoint: null) {
+  [#{$layout}~='#{$horizontal}'] {
+    &[#{$layout}*='#{$align}#{$breakpoint}:#{$vertical-stretch}'] {
+      align-items: stretch;
       align-content: stretch;
+      flex-grow: 1;
     }
 
-    &[#{$layout}*='#{$align}#{$breakpoint}:#{$stretch}'] > *,
-    &[#{$layout}*='#{$align}#{$breakpoint}:#{$horizontal-stretch}'] > *,
-    &[#{$layout}*='#{$align}#{$breakpoint}:#{$stretch}']::slotted(*),
-    &[#{$layout}*='#{$align}#{$breakpoint}:#{$horizontal-stretch}']::slotted(*) {
-      flex-grow: 1 !important;
-    }
-
-    &[#{$layout}*='#{$align}#{$breakpoint}:#{$stretch}'] > *,
-    &[#{$layout}*='#{$align}#{$breakpoint}:#{$vertical-stretch}'],
-    &[#{$layout}*='#{$align}#{$breakpoint}:#{$stretch}']::slotted(*),
-    &[#{$layout}*='#{$align}#{$breakpoint}:#{$vertical-stretch}']::slotted(*) {
-      align-content: stretch !important;
-    }
-
-    &[#{$layout}~='wrap#{$breakpoint}:none'] {
-      flex-wrap: nowrap;
-    }
+    &[#{$layout}*='#{$align}#{$breakpoint}:#{$horizontal-stretch}'] {
+      justify-content: stretch;
+      flex-grow: 1;
 
-    &[#{$layout}~='order#{$breakpoint}:reverse'] {
-      flex-direction: row-reverse;
+      & > *,
+      & > ::slotted(*) {
+        flex-grow: 1;
+      }
     }
 
-    // Align Single Item
-    & > [#{$layout}*='#{$align}#{$breakpoint}:#{$stretch}'],
-    &::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$stretch}']) {
-      flex-grow: 1 !important;
-    }
+    &[#{$layout}*='#{$align}#{$breakpoint}:#{$stretch}'] {
+      align-items: stretch;
+      align-content: stretch;
+      flex-grow: 1;
 
-    & > [#{$layout}*='#{$align}#{$breakpoint}:#{$shrink}'],
-    &::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$shrink}']) {
-      flex-shrink: 1 !important;
-      flex-grow: 0 !important;
+      & > *,
+      & > ::slotted(*) {
+        flex-grow: 1;
+      }
     }
+  }
+}
 
+@mixin generateHorizontalItemAlignment($breakpoint: null) {
+  [#{$layout}~='#{$horizontal}'] {
     & > [#{$layout}*='#{$align}#{$breakpoint}:#{$center}'],
     & > [#{$layout}*='#{$align}#{$breakpoint}:#{$vertical-center}'],
-    &::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$center}']),
-    &::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$vertical-center}']) {
+    & > ::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$center}']),
+    & > ::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$vertical-center}']) {
       align-self: center;
     }
 
     & > [#{$layout}*='#{$align}#{$breakpoint}:#{$center}'],
     & > [#{$layout}*='#{$align}#{$breakpoint}:#{$horizontal-center}'],
-    &::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$center}']),
-    &::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$horizontal-center}']) {
+    & > ::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$center}']),
+    & > ::slotted([#{$layout}*='#{$align}#{$breakpoint}:#{$horizontal-center}']) {
```

Comment:
This makes the slot selector behave much like the regular light DOM `>`. This seems to help improve consistency with the slot behavior in layouts